### PR TITLE
get_all_passwords_as_generator

### PR DESCRIPTION
**Pull request by Hellowlol**
This is untested but what about something like this?

### CODE REVIEW

**Comment by Hellowlol** [comment]
https://github.com/sabnzbd/sabnzbd/issues/877

**Comment by Safihre** [comment]
Maybe test it? :P

**Comment by Hellowlol** [comment]
I wouldn't find a nzb that was encrypted... lol

**Comment by Hellowlol** [comment]
So i have downloaded over 50 random nzbs still no, password protected. I tried to upload one myself, but it does not work, dunno why it dont work. Does anybody have link to a password protected nzb so i can test it im happy to do it.

**Comment by Safihre** [comment]
Here: [Test123{{nZb-by-USC}}.txt](https://github.com/sabnzbd/sabnzbd/files/888276/Test123.nZb-by-USC.txt) (have to rename the extension to `.nzb`)

**Comment by Hellowlol** [comment]
Thanks! I just tested it and it seemed to work just nicely. Ran the same file on develop and it crashed after 10 sec on a memory error (250ish mb password file.)

File: sabnzbd/assembler.py [hunk 1]
```diff
@@ -326,14 +326,14 @@ def check_encrypted_and_unwanted_files(nzo, filepath):
                     if is_cloaked(filepath, zf.namelist()):
                         nzo.encrypted = 1
                         encrypted = True
-                    elif not sabnzbd.HAVE_CRYPTOGRAPHY and not passwords:
```
Comment (Hellowlol) [inline sabnzbd/assembler.py]: I removed this. I dont understand why this is needed. I dont understand the comment it makes no sense to me.

Comment (thezoggy) [inline sabnzbd/assembler.py]: if you dont understand it.. why remove it.

**Comment by Safihre** [comment]
@Hellowlol: thanks for taking the time. I will merge this after 2.0.0-Final so it can be in 2.0.1, but for now we will first probably have an RC3 due to some unforseen bugs. 
The implementation is indeed how I also would have done it and I will take a look again at the Cryptography warning, right now I also don't understand anymore why I did that but I remember it was for a good reason :P

**Comment by Hellowlol** [comment]
@Safihre Thanks. No worries, i dont think the password file feature is used by many, its just simpler to download a new nzb imo. I squashed the pr.

File: sabnzbd/assembler.py [hunk 2]
```diff
@@ -326,14 +327,15 @@ def check_encrypted_and_unwanted_files(nzo, filepath):
...
-                        # if no cryptography installed, only error when no password was set
+
+                    elif not sabnzbd.HAVE_CRYPTOGRAPHY and any(slice(passwords, 2)) is False:
```
Comment (Hellowlol) [inline sabnzbd/assembler.py]: This should be isslice not slice.

**Comment by Safihre** [comment]
This one is a bit messy, I do understand the beauty of a Python generator but this approach does not seem to really increase readability of our code.
I think I would rather implement a warning message if more than 50 passwords are detected. This will inform users that think unrar can handle such things easily.
What do you think?

**Comment by Hellowlol** [comment]
Honestly I think it simpler then the code we have today. I still think this should be a generator it's up to you. I skip all password protected nzbs so I will never need this feature anyway. I just did it since it was a easy fix and I wanted contribution something to sab. If this isn't accepted I'm gonna find some typos to fix :D

**Pull request closed without merging**